include/zephyr/arch/riscv/arch.h

```diff
@@ -220,6 +220,7 @@ extern void z_irq_spurious(const void *unused);
  */
 static ALWAYS_INLINE unsigned int arch_irq_lock(void)
 {
+#ifndef SOC_RISCV_ARCH_IRQ_LOCK
```

Comment:
What would be a reason to not follow the existing `z_soc_irq_...` pattern here?

https://github.com/zephyrproject-rtos/zephyr/blob/abe50cc92991389e2508de8ea721bee36cd5fb17/include/zephyr/arch/arm/aarch32/irq.h#L54-L57

https://github.com/zephyrproject-rtos/zephyr/blob/abe50cc92991389e2508de8ea721bee36cd5fb17/include/zephyr/arch/xtensa/irq.h#L77-L80

https://github.com/zephyrproject-rtos/zephyr/blob/abe50cc92991389e2508de8ea721bee36cd5fb17/include/zephyr/arch/arm64/irq.h#L53-L56

Reply:
Because vendors can do whatever they want with RISC-V and we have to support it. So in the use-case I have in mind the core is shipping a PLIC controller following the spec but a new CSR was added to the ISA that is used to implement critical sections (thus used in the `arch_irq_*` family)

Reply:
I am still not seeing why the function should not be named `z_soc_irq_*` (e.g. `z_soc_irq_lock`), that can be optionally "mapped" to `arch_irq_*` when the relevant Kconfig is enabled (e.g. `CONFIG_RISCV_CUSTOM_IRQ_CSR`).

Relying on `#ifndef SOC_RISCV_ARCH_IRQ_*`, instead of a Kconfig to explicitly control this behaviour is a rather fragile approach.

Also it looks like the return value of `SOC_RISCV_ARCH_IRQ_LOCK` is not properly handled?

Reply:
@stephanosio pushed a new version based on your suggestions